recipes/binutils/all/conanfile.py

```diff
@@ -126,64 +129,78 @@ def _raise_unsupported_configuration(self, key, value):
         raise ConanInvalidConfiguration(f"This configuration is unsupported by this conan recip. Please consider adding support. ({key}={value})")
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path"):
             self.build_requires("msys2/cci.latest")
 
     def requirements(self):
-        self.requires("zlib/1.2.12")
+        self.requires("zlib/1.2.13")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self.source_folder)
 
     @property
     def _exec_prefix(self):
         return os.path.join(self.package_folder, "bin", "exec_prefix")
 
-    @functools.lru_cache(1)
-    def _configure_autotools(self):
-        autotools = AutoToolsBuildEnvironment(self, win_bash=self._settings_build.os == "Windows")
+    def generate(self):
         yes_no = lambda tf : "yes" if tf else "no"
-        conf_args = [
-            f"--target={self.options.target_triplet}",
-            f"--enable-multilib={yes_no(self.options.multilib)}",
-            "--with-system-zlib",
-            "--disable-nls",
-            f"--program-prefix={self.options.prefix}",
-            f"exec_prefix={tools.unix_path(self._exec_prefix)}",
-        ]
-        autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return autotools
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.append("--disable-nls")
+        tc.configure_args.append(f"--target={self.options.target_triplet}")
+        tc.configure_args.append(f"--enable-multilib={yes_no(self.options.multilib)}")
+        tc.configure_args.append(f"--with-zlib={self.deps_cpp_info['zlib'].rootpath}")
+        tc.configure_args.append(f"--program-prefix={self.options.prefix}")
+        tc.configure_args.append(f"--exec_prefix={unix_path(self, os.path.join('/','bin', 'exec_prefix'))}")
```

Comment:
This new syntax 🤢 

This broke my brain for a good 30 seconds -- https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html?highlight=unix_path

***

Why does this need to be root? should it not be relative? the old one was absolute to the package folder https://github.com/conan-io/conan-center-index/pull/14137/files#diff-c68fc5112c8ad8e228d2ea59d42386921fb01b28ab2be48ec5f060a0218a76ffR143

Reply:
I'm not convinced that this was doing anything in the previous version, note the missing `--` prepended to `exec_prefix`. The confusing thing about this line, and the way gnu projects seem to accept paths in general, is that this _is_ a relative path, even though it looks like an absolute path with the leading `/`. If this is set to the `package_folder`, then the `exec_prefix` ends up being something like `/home/user/.conan/data/binutils/2.38/_/_/package/<package id>/`, which gets embedded in the exported package - not ideal.

Long story short - works as intended. I guess given that this would only ever run in a unix environment, even on windows, I guess this could be simplified to
```python
        tc.configure_args.append(f"--exec_prefix=/bin/exec_prefix")
```

I was concerned that path separators could make things awkward on windows, but yeah given that this is a relative path and can only ever be built in a bash environment this is probably safe enough and much more readable.